sematic/types/types/set.py

```diff
@@ -0,0 +1,79 @@
+# Standard Library
+from typing import Any, Iterable, List, Optional, Set, Type, Tuple
+
+# Sematic
+from sematic.types.casting import safe_cast
+from sematic.types.registry import (
+    SummaryOutput,
+    register_from_json_encodable,
+    register_safe_cast,
+    register_to_json_encodable,
+    register_to_json_encodable_summary,
+)
+from sematic.types.serialization import (
+    get_json_encodable_summary,
+    value_from_json_encodable,
+    value_to_json_encodable,
+)
+
+
+@register_safe_cast(set)
```

Comment:
Please don't forget to add a `@register_can_cast` function as well.

Reply:
Added!